research/src/pages/file/file.proposal.mdx

```diff
@@ -0,0 +1,248 @@
+| ------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [`select`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/select) | Returns, as `select()` does not apply to this element ([reference](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-textarea/input-select)) |
+
+## Events
+
+| Event Name | Detail Type | Bubbles | Composed | Cancellable | Dispatch Behavior                                       |
+| ---------- | ----------- | ------- | -------- | ----------- | ------------------------------------------------------- |
+| `change`   | none        | `true`  | `true`   | `false`     | Fired when the input's `value` is commited by the user. |
+| `input`    | none        | `true`  | `true`   | `false`     | Fired when the input's `value` changes.                 |
+
+### Appearance
+
+The following examples are taken from Chrome 84, and are consistent with
+implementations in other browsers.
+
+#### Default
+
+<Image
+  alt="Example of the file input control in Chromium, with no files selected"
+  src="chromium-file-default.png"
+/>
+<br />
+<Image
+  alt="Example of the file input control in Chromium, with files selected"
+  src="chromium-file-default-selection.png"
+/>
+<br />
+<br />
+
+#### Set to `multiple`
+
+<Image
+  alt="Example of the file input control in Chromium, set to multiple, with no files selected"
+  src="chromium-file-multiple.png"
+/>
+<br />
+<Image
+  alt="Example of the file input control in Chromium, set to multiple, with files selected"
+  src="chromium-file-multiple-selection.png"
+/>
+<br />
+<br />
+
+### Anatomy
+
+#### Diagram
+
+<FileAnatomy />
+
+#### DOM Structure
+
+```html
+<host>
+  <div part="button">
+    <slot name="button"></slot>
+  </div>
+  <div part="label">
+    <slot name="label"></slot>
+  </div>
+</host>
+```
+
+#### Slots
+
+| Slot Name | Description                                                                           | Fallback Content              |
+| --------- | ------------------------------------------------------------------------------------- | ----------------------------- |
+| `button`  | Activates the file selection prompt.                                                  | A button element is provided. |
+| `label`   | Indicates if a file input is waiting for file selection, or if it has selected files. | A label element is provided.  |
+
+#### CSS Parts
+
+| Part Name | Description                                           |
+| --------- | ----------------------------------------------------- |
+| `button`  | The controls button.                                  |
+| `label`   | The controls label, indicating file selection status. |
+
+---
+
+## Behavior
+
+### States and Interactions
```

Comment:
By `open`, do you mean that the system's file picker is open?

If so, this is not something I've seen in anything I've come across during research. Possibly because it's not possible to do (if you're not a browser vendor), and because the user is already _trapped_ in the system's file picker until they've selected a file.

Reply:
yes, this is for when the window is open just in case you want to do some variant. Given we are adding the select has an open state that the popup is open seems for consistency reasons to do it here. You're correct in the current world since no file system access that the doc will be frozen by the modal but that doesn't mean that will always be the case. But this can be discussed in an issue rather than plague your PR :)

Reply:
I've raised an issue, what type of research do you have in mind here? I wasn't sure what to add.
https://github.com/WICG/open-ui/issues/181

Reply:
@mrmckeb research would be is there a common paradigm that when a file popup is open that this has implications on other aspects of the scenario? EG: Do icons, style or content change in any meaningful way?

Reply:
As the browser doesn't provide an open state, custom components can't do anything to show this state change.

Native file inputs also provide no visual feedback that the picker is open - apart from the picker itself, which blocks further interaction with the page until closed.